client/client.go

```diff
@@ -460,6 +469,21 @@ func (c *Client) setupDrivers() error {
 	return nil
 }
 
+// setupDiscovery sets up the discovery layers and initializes them.
+func (c *Client) setupDiscovery() error {
+	// Create the discovery layer
+	disc, err := discovery.NewDiscoveryLayer(discovery.Builtins, c.config,
+		c.logger, c.config.Node)
+	if err != nil {
+		return err
+	}
+	c.discovery = disc
+
+	avail := disc.EnabledProviders()
+	c.logger.Printf("[DEBUG] client: available discovery layers: %v", avail)
```

Comment:
Here I was just trying to stay consistent with what drivers and fingerprints do. I can `strings.Join()` them or something, if you think that's better, but we should do the others too in that case.